apps/omg_watcher/test/block_getter/core_test.exs

```diff
@@ -202,80 +204,76 @@ defmodule OMG.Watcher.BlockGetter.CoreTest do
-      |> Core.get_numbers_of_blocks_to_download(3_000)
-
-    potential_withholding = Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 0, 0)
-    assert {:ok, state, []} = Core.handle_downloaded_block(state, potential_withholding)
-
-    potential_withholding = Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 0, 1)
-
-    assert {{:needs_stopping, :withholding}, _, [%Event.BlockWithholding{blknum: 2000}]} =
-             Core.handle_downloaded_block(state, potential_withholding)
+  test "handle_downloaded_block function called twice with PotentialWithholdingReport returns BlockWithholding event" do
+    init_state(opts: [maximum_number_of_pending_blocks: 5, maximum_block_withholding_time_ms: 0])
+    |> Core.get_numbers_of_blocks_to_download(3_000)
+    |> assert_check([1_000, 2_000])
+    |> Core.handle_downloaded_block(Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 0, 0))
+    |> assert_check(:ok, [])
+    |> Core.handle_downloaded_block(Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 0, 1))
+    |> assert_check({:needs_stopping, :withholding}, [%Event.BlockWithholding{blknum: 2000}])
   end
 
   test "get_numbers_of_blocks_to_download function returns number of potential withholding block which then is canceled" do
-    {state, [1_000, 2_000, 3_000, 4_000]} =
-      init_state(opts: [maximum_number_of_pending_blocks: 4, maximum_block_withholding_time_ms: 0])
-      |> Core.get_numbers_of_blocks_to_download(20_000)
-
-    state =
-      state
-      |> handle_downloaded_block(%Block{number: 1_000})
-      |> handle_downloaded_block(%Block{number: 2_000})
-
-    potential_withholding = Core.validate_download_response({:error, :error_reason}, <<>>, 3_000, 0, 0)
-    assert {:ok, state, []} = Core.handle_downloaded_block(state, potential_withholding)
-
-    assert {_, [3000, 5000, 6000]} = Core.get_numbers_of_blocks_to_download(state, 20_000)
-
-    assert {:ok, state, []} = Core.handle_downloaded_block(state, {:ok, %Block{number: 3_000}})
+    init_state(opts: [maximum_number_of_pending_blocks: 4, maximum_block_withholding_time_ms: 0])
+    |> Core.get_numbers_of_blocks_to_download(20_000)
+    |> assert_check([1_000, 2_000, 3_000, 4_000])
+    |> handle_downloaded_block(%Block{number: 1_000})
+    |> handle_downloaded_block(%Block{number: 2_000})
+    |> Core.handle_downloaded_block(Core.validate_download_response({:error, :error_reason}, <<>>, 3_000, 0, 0))
+    |> assert_check(:ok, [])
+    |> Core.get_numbers_of_blocks_to_download(5_000)
+    |> assert_check([3_000])
+    |> handle_downloaded_block(%Block{number: 3_000})
+    |> Core.get_numbers_of_blocks_to_download(20_000)
+    |> assert_check([5_000, 6_000, 7_000])
+  end
 
-    assert {_, [5000, 6000, 7000, 8000]} = Core.get_numbers_of_blocks_to_download(state, 20_000)
+  test "get_numbers_of_blocks_to_download don't send block that are download" do
```

Comment:
get_number_of_blocks_to_download does not return blocks that are being downloaded